### DISCUSSION TITLE
PiVPN for clients

### DISCUSSION THREAD

**Discussion by WayneKeenan** [pivpn/pivpn#1396] [Feature requests]
Is there a script for setting OpenVPN Pi clients on RaspberryPis?
I want to create a hub (PiVPN Server) and spoke  (Pi clients) architecture.

**Comment by jcstill** [comment]
Couldn't help but start to read through some issues while I wait for my keys to generate. (rpi b+, 4096 bits, 10 hours later...)

Try [this](https://openvpn.net/index.php/access-server/docs/admin-guides/182-how-to-connect-to-access-server-with-linux-clients.html)

I haven't tested this as I am currently waiting...

**Comment by WayneKeenan** [comment]
Thanks, that's the underlying VPN software, but my question is fishing for the existence of any 'client' script that wraps OpenVPN as the current PiVPN does for the server side.  e.g. perhaps a PiVPN client script that might be installed such as:

```bash
curl -L https://client-install.pivpn.io | bash
```
Which:
1. does the OpenVPN install
2. asks for some options
3. keeps the VPN connection alive should it disconnect (across reboots)
4. manages to get around the OpenVPN Linux DNS caveat, somehow

It would balance the current PiVPN server setup offering, if such a thing does not currently exist.

If no such thing exists I might get a chance to look at it when I get around to setting up my OpenVPN client.

**Comment by orazioedoardo** [comment]
1) On the server, generate a client config with no password.
2) On the client, install OpenVPN with `sudo apt install openvpn`
3) Copy the .ovpn file to `/etc/openvpn/` and rename it as `whatever.conf`.
4) Run `sudo chown root:root /etc/openvpn/whatever.conf`
5) Run `sudo chmod 700 /etc/openvpn/whatever.conf`
6) Run `sudo systemctl start openvpn@whatever.service`
7) Run `sudo systemctl enable openvpn@whatever.service`

Check if the public IP is the one from the raspberry server with: `dig +short myip.opendns.com @resolver1.opendns.com`. If it works, reboot and check again.

**Reply by viertelb** [reply]
> 1. On the server, generate a client config with no password.
> 
>     2. On the client, install OpenVPN with `sudo apt install openvpn`
> 
>     3. Copy the .ovpn file to `/etc/openvpn/` and rename it as `whatever.conf`.
> 
>     4. Run `sudo chown root:root /etc/openvpn/whatever.conf`
> 
>     5. Run `sudo chmod 700 /etc/openvpn/whatever.conf`
> 
>     6. Run `sudo systemctl start openvpn@whatever.service`
> 
>     7. Run `sudo systemctl enable openvpn@whatever.service`
> 
> 
> Check if the public IP is the one from the raspberry server with: `dig +short myip.opendns.com @resolver1.opendns.com`. If it works, reboot and check again. 

Is this still valid? After running the `pivpn` installation a `CLIENT.conf` was generated on my server. It contains of the following entries. I copied its content to `/etc/openvpn/CLIENT.conf` and `sudo systemctl start openvpn@CLIENT.service` gave the following error in `journalctl -xe`:

```
Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/oldpi.conf:1: [Interface] (2.5.1)
```

The config is this. I tried around and it seems no option in that file is recognised by `openvpn`. I set it up with `WireGuard` but the client should be able to connect via `openvpn`, right?

```
[Interface]
PrivateKey = KEY=
Address = 10.6.187.2/24
DNS = 9.9.9.9, 149.112.112.112

[Peer]
PublicKey = KEY=
PresharedKey = KEY=
Endpoint = DYNDNS:51820
AllowedIPs = 0.0.0.0/0, ::0/0
```

**Comment by WayneKeenan** [comment]
awesome, thanks very much!
